### PR TITLE
feat: move axon-tools to axon

### PR DESCRIPTION
**Pull request by wenyuanhust**
## What this PR does / why we need it?

This PR 
- moves the std part of axon-tools into the devtools/axon-tools of axon.
- add tests for block verify and tx verify. 

### What is the impact of this PR?

No Breaking Change

**PR relation**:
- Ref #

<details><summary>CI Settings</summary><br/>

### **CI Usage**

**Tip**: Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [ ] Web3 Compatible Tests
- [ ] OCT 1-5 And 12-15
- [ ] OCT 6-10
- [ ] OCT 11
- [ ] OCT 16-19
- [ ] v3 Core Tests

### **CI Description**

| CI Name                                   | Description                                                               |
| ----------------------------------------- | ------------------------------------------------------------------------- |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                                       |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3                        |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin                      |

</details>

### CODE REVIEW

File: devtools/axon-tools/src/types.rs [hunk 1]
```diff
@@ -0,0 +1,815 @@
+use std::cmp::Ordering;
```
Comment (KaoImin) [inline devtools/axon-tools/src/types.rs]:
```suggestion
use core::cmp::Ordering;
```

File: devtools/axon-tools/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
+    InvalidAddressSourceType,
+
+    #[display(fmt = "Missing interoperation sender")]
+    MissingInteroperationSender,
+
+    #[display(fmt = "InvalidBlockVersion {:?}", _0)]
+    InvalidBlockVersion(u8),
+}
+
+impl std::error::Error for TypesError {}
```
Comment (KaoImin) [inline devtools/axon-tools/src/types.rs]: The `TypesError` should be moved to `error.rs`.

**Review by KaoImin — COMMENTED** [review]
The rlp `Encodable` and `Decodable` implement code should be abstract into a standalone`rlp_codec.rs` file.

File: devtools/axon-tools/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,741 @@
...
+impl Encodable for Vote {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        let vote_type: u8 = self.vote_type;
+        s.begin_list(4)
+            .append(&self.height)
+            .append(&self.round)
+            .append(&vote_type)
+            .append(&self.block_hash.to_vec());
+    }
+}
```
Comment (KaoImin) [inline devtools/axon-tools/src/types.rs]: The code should move to `rlp_codec.rs`.

**Review by KaoImin — DISMISSED** [review]

File: devtools/axon-tools/src/tests/verify_proof.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
+#[cfg(test)]
+mod tests {
```
Comment (KaoImin) [inline devtools/axon-tools/src/tests/verify_proof.rs]: No need to create a `tests` module in `tests` module.

File: devtools/axon-tools/src/tests/verify_trie_proof.rs [hunk 5]
```diff
@@ -0,0 +1,176 @@
+#[cfg(test)]
+mod tests {
```
Comment (KaoImin) [inline devtools/axon-tools/src/tests/verify_trie_proof.rs]: ditto

**Review by KaoImin — APPROVED** [review]

**Review by ashuralyk — APPROVED** [review]

**Pull request merged**